src/python/arcor2_arserver/rpc/scene.py

```diff
@@ -181,9 +181,9 @@ async def list_scenes_cb(req: srpc.s.ListScenes.Request, ui: WsClient) -> srpc.s
 @no_project
 async def add_object_to_scene_cb(req: srpc.s.AddObjectToScene.Request, ui: WsClient) -> None:
 
-    assert glob.SCENE
+    async with ctx_write_lock(glob.SCENE.id, 'ui.owner'):
```

Comment:
Jako identifikátor uživatele by šlo jednoduše použít objekt `ui`. Ale mělo by to tu nevýhodu, že kdyby uživatel na chvíli odložil tablet a znovu ho vzal (po uspání tabletu se appka odpojí od serveru), přišel by o svoje zámky. Takže by asi bylo lepší kdyby ui po připojení k serveru poslalo nějaký login request. Asi by stačil prostě nickname (unikátní). Některé věci (získání seznamu objektů, vylistování scén, projektů apod) by mohly fungovat i bez toho loginu. Ale jakmile by ui/uživatel chtěl dělat něco co mění stav na serveru, musel by se přihlásit. Ideálně asi udělat dekorátor, který by to kontroloval. Co myslíte? Neprobíral jste to třeba už nějak s Michalem Kapinusem?

Reply:
Toto jsem zatím neřešil, ale určitě to mám v plánu. Teď to vyplňuji jako placeholder, protože bych to chtěl po dokončení zamykání řešit jako celek (nějaké notifikace se jmény, kdo blokuje zámky atd)